src/components/add-movie-form/AddMovieForm.jsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { useNavigate } from "react-router-dom";
+import Dialog from "../dialog/Dialog"
+import MovieForm from "../movie-form/MovieForm";
+
+export default function AddMovieForm() {
+  const navigate = useNavigate();
+
+  const handleSubmit = async (formData) => {
+    
+    try {
+      const response = await fetch("http://localhost:4000/movies", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+      const result = await response.json();
+      
+      navigate(`/${result.id}`);
+    } catch (error) {
+      console.error("Error adding movie:", error);
+    }
+  };
+
+  const onClose = () => {
+    navigate(`/`);
+  }
+
+  return (
+    <Dialog title="Add Movie" onCloseDialog={onClose}>
+      <MovieForm movie={{}} onSubmit={handleSubmit} />
```

Comment:
We can make this optional according to the requirements

> The MovieForm component should accept the following props:
- an optional object for initial movie info.

And then use a default value when destructuring props

`({ movie = {}, onSubmit })`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#default_value